From 07705cbfb6743b8ea1900cfd8cea6eaed0a1ed9e Mon Sep 17 00:00:00 2001
From: "qiyao.zhou" <qiyao.zhou@amlogic.com>
Date: Fri, 20 Dec 2019 16:30:01 +0800
Subject: [PATCH] tvserver: video don't display corectly. [1/1]

PD#NULL

Problem:
don't set /sys/class/video/disable_video node while start TV.

Solution:
Add set node action in start TV.

Verify:
T962X3(AB301)

Change-Id: I899a2ed2e8bc1e6a646d232fce07456713613736
Signed-off-by: qiyao.zhou <qiyao.zhou@amlogic.com>
---
 libtv/CTv.cpp | 10 ++++++++++
 libtv/CTv.h   | 11 ++++++++++-
 2 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/libtv/CTv.cpp b/libtv/CTv.cpp
index ff05d44..3116851 100644
--- a/libtv/CTv.cpp
+++ b/libtv/CTv.cpp
@@ -60,6 +60,7 @@ int CTv::StartTv(tv_source_input_t source)
 {
     LOGD("%s: source = %d!\n", __FUNCTION__, source);
     int ret = -1;
+    setVideoLayerStatus(VIDEO_LAYER_STATUS_ENABLE_AND_CLEAN);
     tvin_port_t source_port = mpTvin->Tvin_GetSourcePortBySourceInput(source);
     ret = mpTvin->Tvin_OpenPort(source_port);
     mCurrentSource = source;
@@ -262,6 +263,15 @@ int CTv::sendTvEvent(CTvEvent &event)
     return 0;
 }
 
+int CTv::setVideoLayerStatus(videolayer_status_t status)
+{
+    LOGD("%s: status = %d\n", __FUNCTION__, status);
+
+    char temp[8] = {0};
+    sprintf(temp, "%d", status);
+    return tvWriteSysfs(VIDEO_DISABLE_VIDEO, temp);
+}
+
 #ifdef HAVE_AUDIO
 int CTv::mapSourcetoAudiotupe(tv_source_input_t dest_source)
 {
diff --git a/libtv/CTv.h b/libtv/CTv.h
index 61d19ed..c8c3eb7 100644
--- a/libtv/CTv.h
+++ b/libtv/CTv.h
@@ -20,6 +20,14 @@
 
 #define HDMI_EDID14_FILE_PATH    "/vendor/etc/tvconfig/hdmi/port_14.bin"
 #define HDMI_EDID20_FILE_PATH    "/vendor/etc/tvconfig/hdmi/port_20.bin"
+#define VIDEO_DISABLE_VIDEO      "/sys/class/video/disable_video"
+
+typedef enum video_layer_status_e {
+    VIDEO_LAYER_STATUS_ENABLE,
+    VIDEO_LAYER_STATUS_DISABLE,
+    VIDEO_LAYER_STATUS_ENABLE_AND_CLEAN,
+    VIDEO_LAYER_STATUS_MAX,
+} videolayer_status_t;
 
 class CTv : public CTvDevicesPollDetect::ISourceConnectObserver {
 public:
@@ -48,7 +56,8 @@ private:
     void onSigToUnstable();
     void onSigToUnSupport();
     void onSigToNoSig();
-    int  sendTvEvent(CTvEvent &event);
+    int sendTvEvent(CTvEvent &event);
+    int setVideoLayerStatus(videolayer_status_t status);
 #ifdef HAVE_AUDIO
     int mapSourcetoAudiotupe(tv_source_input_t dest_source);
 #endif
-- 
2.7.4

